### PR TITLE
Support account locking feature for email OTP

### PR DESCRIPTION
**Pull request by AnuradhaSK**
## Purpose
Support account locking for email-otp when configured as second factor 
Backport #79

### CODE REVIEW

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java [hunk 1]
```diff
@@ -1605,4 +1628,249 @@ protected void triggerEvent(String userName, String tenantDomain, String userSto
...
+        UserRealm userRealm = null;
+        try {
+            if (authenticatedUser != null) {
+                String tenantDomain = authenticatedUser.getTenantDomain();
+                int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+                RealmService realmService = IdentityTenantUtil.getRealmService();
+                userRealm = realmService.getTenantUserRealm(tenantId);
+            }
+        } catch (UserStoreException e) {
+            throw new AuthenticationFailedException("Cannot find the user realm.", e);
```
Comment (sachiniWettasinghe) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java]: Improve this by adding user info

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java [hunk 2]
```diff
@@ -1605,4 +1628,249 @@ protected void triggerEvent(String userName, String tenantDomain, String userSto
...
+    private void setUserClaimValues(AuthenticatedUser authenticatedUser, Map<String, String> updatedClaims)
+            throws AuthenticationFailedException {
+
+        try {
+            UserRealm userRealm = getUserRealm(authenticatedUser);
+            UserStoreManager userStoreManager = userRealm.getUserStoreManager();
+            userStoreManager.setUserClaimValues(IdentityUtil.addDomainToName(authenticatedUser.getUserName(),
+                    authenticatedUser.getUserStoreDomain()), updatedClaims, UserCoreConstants.DEFAULT_PROFILE);
+        } catch (UserStoreException e) {
+            log.error("Error while updating user claims", e);
```
Comment (sachiniWettasinghe) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java]: Improve the error log by adding user info

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java [hunk 3]
```diff
@@ -1605,4 +1628,249 @@ protected void triggerEvent(String userName, String tenantDomain, String userSto
...
+        try {
+            UserRealm userRealm = getUserRealm(authenticatedUser);
+            UserStoreManager userStoreManager = userRealm.getUserStoreManager();
+            claimValues = userStoreManager.getUserClaimValues(IdentityUtil.addDomainToName(
+                    authenticatedUser.getUserName(), authenticatedUser.getUserStoreDomain()), new String[]{
+                            EmailOTPAuthenticatorConstants.EMAIL_OTP_FAILED_ATTEMPTS_CLAIM,
+                            EmailOTPAuthenticatorConstants.FAILED_LOGIN_LOCKOUT_COUNT_CLAIM},
+                    UserCoreConstants.DEFAULT_PROFILE);
+        } catch (UserStoreException e) {
+            log.error("Error while reading user claims.", e);
```
Comment (sachiniWettasinghe) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java]: Improve the error log by adding user info

Comment (AnuradhaSK) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java]: Fixed in https://github.com/wso2-extensions/identity-outbound-auth-email-otp/pull/86/commits/1ac872b7efc2fbcba59d9df20593b71ef5cd9b60

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java (hunk 2 above)
Comment (AnuradhaSK) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java]: Fixed in 1ac872b

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java (hunk 1 above)
Comment (AnuradhaSK) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java]: Fixed in 1ac872b

**Review by sachiniWettasinghe — APPROVED** [review]

**Pull request merged**